cmake/OpenCVDetectOpenCL.cmake

```diff
@@ -77,4 +77,21 @@ if(OPENCL_FOUND)
       list(APPEND OPENCL_INCLUDE_DIRS "${CLAMDBLAS_INCLUDE_DIR}")
     endif()
   endif()
+
+  if(WITH_OPENCLNVEXT)
+    find_path(CLNVEXT_INCLUDE_DIR
+      NAMES CL/cl_d3d11_ext.h
+      PATHS ENV CUDA_PATH ENV ProgramFiles
+      PATH_SUFFIXES include
+      DOC "clNvExt include directory")
+
+    if(CLNVEXT_INCLUDE_DIR)
+      set(HAVE_CLNVEXT on)
+      list(FIND OPENCL_INCLUDE_DIRS "${CLNVEXT_INCLUDE_DIR}" FIND_OUT)
+      if (FIND_OUT MATCHES -1)
+        list(APPEND OPENCL_INCLUDE_DIRS "${CLNVEXT_INCLUDE_DIR}")
```

Comment:
Perhaps this would introduce mess with OpenCL headers from **multiple** include directories.

Check `OPENCL_INCLUDE_DIR` path only for `cl_d3d11_ext.h` header.

So your usage should be:
```
-DOPENCL_INCLUDE_DIR=<CUDA_PATH>/include
-DWITH_OPENCL_D3D11_NV=ON
```

Reply:
OK